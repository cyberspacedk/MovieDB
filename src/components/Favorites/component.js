/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import React from 'react';
import { Layout, Row, Col, Pagination, BackTop, Icon, Modal } from 'antd';
import Spinner from '../shared/StatusFields/Spinner';
import Error from '../shared/StatusFields/Error';
import Empty from '../shared/StatusFields/Empty';
import Card from '../shared/Card';

const Favorites = ({
  favoritesList,
  isLoading,
  isError,
  history,
  removeFav,
  currentPage,
  totalPage,
}) => {
  const removeFavModal = (e, id) => {
    e.stopPropagation();
    Modal.confirm({
      title: 'Do you want to delete movie from favorites?',
      onOk() {
        removeFav(id, false);
      },
      onCancel() {},
    });
  };

  return (
    <Layout>
      <Layout.Content>
        <div className="top-margin">
          <Row type="flex" gutter={16} justify="center">
            <Col span={20}>
              {isLoading && <Spinner />}
              {isError && <Error />}
              {!favoritesList.length && <Empty />}
              {favoritesList &&
                favoritesList.map(item => (
                  <Card
                    key={item.id}
                    id={item.id}
                    poster={item.poster_path}
                    title={item.original_title}
                    overview={item.overview}
                    history={history}
                    actions={[
                      <Icon
                        key="delete"
                        type="delete"
                        onClick={e => removeFavModal(e, item.id)}
                      />,
                    ]}
                  />
                ))}
            </Col>
          </Row>

          {totalPage ? (
            <Row type="flex" justify="center">
              <Col>
                <Pagination
                  defaultCurrent={1}
                  total={totalPage}
                  className="pagination"
                  defaultPageSize={20}
                />
              </Col>
            </Row>
          ) : null}
        </div>
      </Layout.Content>
      <BackTop />
    </Layout>
  );
};

export default Favorites;
