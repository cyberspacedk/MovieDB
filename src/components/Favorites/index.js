import FavoritesContainer from './container';

export default FavoritesContainer;
