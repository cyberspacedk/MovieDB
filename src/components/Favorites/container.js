/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Favorites from './component';
import {
  getFavoritesRequest,
  operationsFavoritesRequest,
} from '../../store/favorites/actions';
import {
  getFavoritesList,
  isError,
  isLoading,
  isEmpty,
  getTotalPages,
} from '../../store/favorites/selectors';

class FavoritesContainer extends Component {
  componentDidMount() {
    this.props.getFavoritesRequest();
  }

  render() {
    return <Favorites {...this.props} />;
  }
}

FavoritesContainer.propTypes = {
  getFavoritesRequest: PropTypes.func.isRequired,
};

const mstp = state => ({
  favoritesList: getFavoritesList(state),
  loading: isLoading(state),
  error: isError(state),
  empty: isEmpty(state),
  totalPage: getTotalPages(state),
});

const mdtp = {
  getFavoritesRequest,
  operationsFavoritesRequest,
};

export default connect(
  mstp,
  mdtp,
)(FavoritesContainer);
