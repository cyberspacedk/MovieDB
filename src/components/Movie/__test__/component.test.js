import React from 'react';
import { shallow } from 'enzyme';
import Favorites from '../component';

describe('Component: Favorites', () => {
  describe('Status fields. Error , Loading', () => {
    const props = {
      error: true,
      loading: false,
      aboutFilm: {},
    };

    it('Match its snapshot. Error', () => {
      const wrapper = shallow(<Favorites {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Match its snapshot. Loading', () => {
      const nextProps = {
        ...props,
        error: false,
        loading: true,
      };
      const wrapper = shallow(<Favorites {...nextProps} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Match its snapshot. About film object provided', () => {
    const props = {
      error: false,
      loading: false,
      aboutFilm: {
        id: 1001,
        original_title: 'Robocop',
        overview: 'Dead cop kill his murders',
        original_language: 'en',
        runtime: 150,
        budget: 7000000,
        revenue: 90005,
        backdrops: [{ file_path: 'path' }],
        genres: [{ id: 1, name: 'Action' }],
        cast: [
          { cast_id: 1, profile_path: 'path', name: 'Bill', character: 'Cop' },
        ],
        crew: [
          {
            credit_id: 177,
            profile_path: 'path',
            name: 'John',
            department: '3D',
          },
        ],
      },
    };
    it('Should draw component that shows film data', () => {
      const wrapper = shallow(<Favorites {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Match its snapshot. About film object provided.Another lang', () => {
      const nextProps = {
        ...props,
        aboutFilm: {
          ...props.aboutFilm,
          original_language: 'ua',
        },
      };
      const wrapper = shallow(<Favorites {...nextProps} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
