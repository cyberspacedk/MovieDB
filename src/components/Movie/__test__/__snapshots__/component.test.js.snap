// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Favorites Match its snapshot. About film object provided Match its snapshot. About film object provided.Another lang 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
      type="flex"
    >
      <Col
        span={24}
      >
        <Carousel
          arrows={false}
          dots={true}
          draggable={false}
        >
          <div
            key="path"
          >
            <img
              alt="Poster"
              className="movie-image"
              src="https://image.tmdb.org/t/p/originalpath"
            />
          </div>
        </Carousel>
      </Col>
    </Row>
    <div
      className="top-margin"
    >
      <Row
        gutter={0}
      >
        <Col
          offset={2}
          span={20}
        >
          <Title
            level={1}
          >
            Robocop
            <Connect(CreateListActionContainer)
              movieId={1001}
            />
            <Connect(FavoriteActionContainer)
              movieId={1001}
            />
            <Connect(WatchListActionContainer)
              movieId={1001}
            />
          </Title>
          <Title
            level={3}
          >
            Overview
          </Title>
          <Paragraph>
            Dead cop kill his murders
          </Paragraph>
        </Col>
      </Row>
      <Row
        gutter={0}
      >
        <Col
          offset={2}
          span={20}
        >
          <Paragraph>
            <span
              className="accent-movie"
            >
              Original Language : 
            </span>
            Europian
          </Paragraph>
        </Col>
        <Col
          offset={2}
          span={20}
        >
          <Paragraph>
            <span
              className="accent-movie"
            >
              Runtime :
            </span>
            2 h 30 m
          </Paragraph>
        </Col>
        <Col
          offset={2}
          span={20}
        >
          <Paragraph>
            <span
              className="accent-movie"
            >
              Budget :
            </span>
            7,000,000.00
          </Paragraph>
        </Col>
        <Col
          offset={2}
          span={20}
        >
          <Paragraph>
            <span
              className="accent-movie"
            >
              Revenue :
            </span>
            90,005.00
          </Paragraph>
        </Col>
        <Col
          offset={2}
          span={20}
        >
          <Paragraph>
            <span
              className="accent-movie"
            >
              Genres :
            </span>
            <Tag
              closable={false}
              key="1"
            >
              ACTION
            </Tag>
          </Paragraph>
        </Col>
      </Row>
      <Row
        gutter={0}
      >
        <Col
          className="top-margin"
          offset={2}
          span={10}
        >
          <Title
            level={3}
          >
            <span
              className="accent-movie"
            >
              Casts
            </span>
          </Title>
        </Col>
      </Row>
      <Row
        gutter={8}
        type="flex"
      >
        <Col
          offset={2}
          span={20}
        >
          <Col
            key="1"
            lg={
              Object {
                "span": 4,
              }
            }
            md={
              Object {
                "span": 6,
              }
            }
            sm={
              Object {
                "span": 8,
              }
            }
            xl={
              Object {
                "span": 4,
              }
            }
            xs={
              Object {
                "span": 12,
              }
            }
          >
            <Card
              className="casts-card"
              cover={
                <img
                  alt="Person"
                  className="person-image"
                  src="https://image.tmdb.org/t/p/w200path"
                />
              }
            >
              <Meta
                description="Cop"
                title="Bill"
              />
            </Card>
          </Col>
        </Col>
      </Row>
      <Row
        gutter={0}
      >
        <Col
          className="top-margin"
          offset={2}
          span={10}
        >
          <Title
            level={3}
          >
            <span
              className="accent-movie"
            >
              Crew
            </span>
          </Title>
        </Col>
      </Row>
      <Row
        gutter={8}
        type="flex"
      >
        <Col
          offset={2}
          span={20}
        >
          <Col
            key="177"
            lg={
              Object {
                "span": 4,
              }
            }
            md={
              Object {
                "span": 6,
              }
            }
            sm={
              Object {
                "span": 8,
              }
            }
            xl={
              Object {
                "span": 4,
              }
            }
            xs={
              Object {
                "span": 12,
              }
            }
          >
            <Card
              className="casts-card"
              cover={
                <img
                  alt="Person"
                  className="person-image"
                  src="https://image.tmdb.org/t/p/w200path"
                />
              }
            >
              <Meta
                description="3D"
                title="John"
              />
            </Card>
          </Col>
        </Col>
      </Row>
    </div>
  </Adapter>
  <BackTop
    visibilityHeight={400}
  />
</Adapter>
`;

exports[`Component: Favorites Match its snapshot. About film object provided Should draw component that shows film data 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
      type="flex"
    >
      <Col
        span={24}
      >
        <Carousel
          arrows={false}
          dots={true}
          draggable={false}
        >
          <div
            key="path"
          >
            <img
              alt="Poster"
              className="movie-image"
              src="https://image.tmdb.org/t/p/originalpath"
            />
          </div>
        </Carousel>
      </Col>
    </Row>
    <div
      className="top-margin"
    >
      <Row
        gutter={0}
      >
        <Col
          offset={2}
          span={20}
        >
          <Title
            level={1}
          >
            Robocop
            <Connect(CreateListActionContainer)
              movieId={1001}
            />
            <Connect(FavoriteActionContainer)
              movieId={1001}
            />
            <Connect(WatchListActionContainer)
              movieId={1001}
            />
          </Title>
          <Title
            level={3}
          >
            Overview
          </Title>
          <Paragraph>
            Dead cop kill his murders
          </Paragraph>
        </Col>
      </Row>
      <Row
        gutter={0}
      >
        <Col
          offset={2}
          span={20}
        >
          <Paragraph>
            <span
              className="accent-movie"
            >
              Original Language : 
            </span>
            English
          </Paragraph>
        </Col>
        <Col
          offset={2}
          span={20}
        >
          <Paragraph>
            <span
              className="accent-movie"
            >
              Runtime :
            </span>
            2 h 30 m
          </Paragraph>
        </Col>
        <Col
          offset={2}
          span={20}
        >
          <Paragraph>
            <span
              className="accent-movie"
            >
              Budget :
            </span>
            7,000,000.00
          </Paragraph>
        </Col>
        <Col
          offset={2}
          span={20}
        >
          <Paragraph>
            <span
              className="accent-movie"
            >
              Revenue :
            </span>
            90,005.00
          </Paragraph>
        </Col>
        <Col
          offset={2}
          span={20}
        >
          <Paragraph>
            <span
              className="accent-movie"
            >
              Genres :
            </span>
            <Tag
              closable={false}
              key="1"
            >
              ACTION
            </Tag>
          </Paragraph>
        </Col>
      </Row>
      <Row
        gutter={0}
      >
        <Col
          className="top-margin"
          offset={2}
          span={10}
        >
          <Title
            level={3}
          >
            <span
              className="accent-movie"
            >
              Casts
            </span>
          </Title>
        </Col>
      </Row>
      <Row
        gutter={8}
        type="flex"
      >
        <Col
          offset={2}
          span={20}
        >
          <Col
            key="1"
            lg={
              Object {
                "span": 4,
              }
            }
            md={
              Object {
                "span": 6,
              }
            }
            sm={
              Object {
                "span": 8,
              }
            }
            xl={
              Object {
                "span": 4,
              }
            }
            xs={
              Object {
                "span": 12,
              }
            }
          >
            <Card
              className="casts-card"
              cover={
                <img
                  alt="Person"
                  className="person-image"
                  src="https://image.tmdb.org/t/p/w200path"
                />
              }
            >
              <Meta
                description="Cop"
                title="Bill"
              />
            </Card>
          </Col>
        </Col>
      </Row>
      <Row
        gutter={0}
      >
        <Col
          className="top-margin"
          offset={2}
          span={10}
        >
          <Title
            level={3}
          >
            <span
              className="accent-movie"
            >
              Crew
            </span>
          </Title>
        </Col>
      </Row>
      <Row
        gutter={8}
        type="flex"
      >
        <Col
          offset={2}
          span={20}
        >
          <Col
            key="177"
            lg={
              Object {
                "span": 4,
              }
            }
            md={
              Object {
                "span": 6,
              }
            }
            sm={
              Object {
                "span": 8,
              }
            }
            xl={
              Object {
                "span": 4,
              }
            }
            xs={
              Object {
                "span": 12,
              }
            }
          >
            <Card
              className="casts-card"
              cover={
                <img
                  alt="Person"
                  className="person-image"
                  src="https://image.tmdb.org/t/p/w200path"
                />
              }
            >
              <Meta
                description="3D"
                title="John"
              />
            </Card>
          </Col>
        </Col>
      </Row>
    </div>
  </Adapter>
  <BackTop
    visibilityHeight={400}
  />
</Adapter>
`;

exports[`Component: Favorites Status fields. Error , Loading Match its snapshot. Error 1`] = `
<Adapter>
  <Error />
  <BackTop
    visibilityHeight={400}
  />
</Adapter>
`;

exports[`Component: Favorites Status fields. Error , Loading Match its snapshot. Loading 1`] = `
<Adapter>
  <Spinner />
  <BackTop
    visibilityHeight={400}
  />
</Adapter>
`;
